src/main/java/de/hpi/bpt/argos/parsing/util/ArtifactBatchImpl.java

```diff
@@ -0,0 +1,95 @@
+package de.hpi.bpt.argos.parsing.util;
+
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.PersistenceArtifact;
+import de.hpi.bpt.argos.util.performance.WatchImpl;
+import de.hpi.bpt.argos.util.performance.WatchTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class ArtifactBatchImpl implements ArtifactBatch {
+	private static final Logger logger = LoggerFactory.getLogger(ArtifactBatchImpl.class);
+
+	private static final int BATCH_SIZE = 500;
+
+	private WatchTask storageTask;
+	private List<PersistenceArtifact> artifactsToStore;
+	private long totalArtifactsStored;
+
+	/**
+	 * This constructor initializes all members with their default value.
+	 */
+	public ArtifactBatchImpl() {
+		artifactsToStore = new ArrayList<>();
+		totalArtifactsStored = 0;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setup() {
+		storageTask = WatchImpl.start("storing artifacts", () -> {});
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00108)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') '{' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') '}' is not preceded with whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)